aqt/mediasrv.py

```diff
@@ -121,11 +132,16 @@ def log_message(self, format, *args):
                           format%args))
 
     # catch /_anki references and rewrite them to web export folder
+    # catch /_addons references and rewrite them to addons folder
     def _redirectWebExports(self, path):
         targetPath = os.path.join(os.getcwd(), "_anki", "")
         if path.startswith(targetPath):
             newPath = os.path.join(_exportFolder, path[len(targetPath):])
             return newPath
+        targetPath = os.path.join(os.getcwd(), "_addons", "")
+        if self._addonFolder and path.startswith(targetPath):
+            newPath = os.path.join(self._addonFolder, path[len(targetPath):])
+            return newPath
```

Comment:
Not sure allowing indiscriminate access to the add-on folders is wise - users may have stored credentials there for example. Perhaps limit it to a web/ subfolder?

Reply:
My thought process was that this wouldn't change the security model for local add-ons, but I assume your main concern lies with access from outside bad actors / rogue websites, right? If so, I hadn't considered that.

Limiting access to a web subfolder within each add-on package sounds good to me. However, I would like to find a way to preserve access to the user_files directory as well. This could be quite important for add-ons with customizable images or stylesheets. The Visual Feedback add-on allows users to set custom images for failing and passing grades, for instance.

However, it's clear that user_files is the most likely directory to contain sensitive information such as login credentials.

So perhaps going with the following access rules might be an option:

- allow indiscriminate access to a top-level web directory within each add-on folder
- allow access to stylesheets and images found in the user_files folder

Would you be OK with that rule-set?

Reply:
Yes, I'm imagining that a shared deck could be constructed to harvest the login details from specific add-ons and ship them off to a third party if indiscriminate access were allowed.

Re user_files, I worry that it may not meet the needs of future use cases - some add-ons may want to allow user js files as well for example.

Just throwing another idea out - what about allowing add-ons to specify an optional handler instead? By default requests to /_addons/xxxx/path/to/file would do nothing, but if add-on xxxx had registered a media handler (eg mw.addonManager.setWebHandler(__name__, callbackFunc), it is called with "path/to/file", and it returns a bool based on whether to allow the file access or not.

Reply:
Sorry about the late reply, Damien. I didn't have a lot of the time these past couple of days.

That's a good point. We should definitely take future needs like that into consideration.

I really like the idea of add-ons determining access to their own files. However, instead of setting a handler, may I suggest using RegEx rules instead? I think that would avoid a lot of the boilerplate code on the add-on side and it should be flexible enough to cover all common use cases.

I was thinking of an API like `mw.addonManager.setWebExports(__name__, regex_str)`. E.g.: `setWebExports(__name__, r"user_files/.+\.(js|css|png)$")`

Reply:
And perhaps we could also add an optional keyword argument to that to still define a handler function which could be preferentially used instead of matching against the RegEx. Though I can't think of any specific use case for that right now.

Reply:
`setWebExportsHandler(__name__, lambda p: p.startswith("user_files/") and os.path.splitext(p)[1] in (".js", ".css", ".png")` is moderately more explicit, but I'm not sure I'd say it's "a lot of boilerplate" :-)

That said, I don't have a strong preference, and regexs may be a little easier for the common case, so either way works for me.

Reply:
Yeah, I might have oversold that a tad bit :)

OK, I'll start reworking the PR with what we've discussed so far.